src/testing/building-apps.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Building Flutter Apps
+---
+
+The Flutter tooling supports three modes when compiling your app,
+and a headless mode for testing.
+This doc explains the three modes and tells you when to use which.
+For more information on headless testing, see
+[Unit testing.](/testing#unit-testing)
+
+# Flutter modes
+
+You choose the compilation mode depending on where you are in
+the development cycle. Are you debugging your code? Do you
+need profiling information? Are you ready to deploy your app?
+
+The following describes each mode and when to use it.
+
+## Debug
+
+In _debug mode_, the app is set up for debugging on the physical
+device, emulator, or simulator. Debug mode means that:
+
+* [Assertions](https://www.dartlang.org/guides/language/language-tour#assert)
+   are enabled.
+* [Observatory](https://dart-lang.github.io/observatory/) is enabled,
+   allowing you to profile timeline data.
```

Comment:
i'd say "allowing you to use the dart debugger" or some such. profiling timeline data in debug mode is a terrible idea. that's what profile mode is for.

Reply:
Good point